tabular/src/autogluon/tabular/models/tabpfnmix/_internal/core/trainer_finetune.py

```diff
@@ -0,0 +1,286 @@
+            prediction_metrics = PredictionMetrics.from_prediction(y_pred, y_true, self.cfg.task, metric=self.stopping_metric)
+            return prediction_metrics
+        else:
+            return None
+
+    def test_epoch(self, dataloader: torch.utils.data.DataLoader, y_test: np.ndarray) -> PredictionMetrics:
+        # FIXME: test_epoch might be better if it uses the for loop logic with n_ensembles
+        y_hat = self.predict_epoch(dataloader)
+        y_hat_finish = self.y_transformer.inverse_transform(y_hat)
+
+        prediction_metrics = PredictionMetrics.from_prediction(y_hat_finish, y_test, self.cfg.task, metric=self.stopping_metric)
+        return prediction_metrics
+
+    def predict(self, x_support: np.ndarray, y_support: np.ndarray, x_query: np.ndarray) -> np.ndarray:
+        """
+        Give a prediction for the query set.
+        """
+
+        x_support = self.preprocessor.transform(x_support)
+        x_query = self.preprocessor.transform(x_query)
+
+        dataset = DatasetFinetune(
+            self.cfg, 
+            x_support = x_support, 
+            y_support = y_support, 
+            x_query = x_query,
+            y_query = None,
+            max_samples_support = self.cfg.hyperparams['max_samples_support'],
+            max_samples_query = self.cfg.hyperparams['max_samples_query'],
+        )
+
+        loader = self.make_loader(dataset, training=False)
+
+        y_hat_ensembles = []
+
+        for _ in range(self.cfg.hyperparams['n_ensembles']):
+            y_hat = self.predict_epoch(loader)
+            y_hat_ensembles.append(y_hat)
+
+        y_hat_ensembled = sum(y_hat_ensembles) / self.cfg.hyperparams['n_ensembles']
+        y_hat_finish = self.y_transformer.inverse_transform(y_hat_ensembled)
+
+        return y_hat_finish
+
+    def predict_epoch(self, dataloader: torch.utils.data.DataLoader) -> np.ndarray:
+        """
+        Returns the predictions for the data in the dataloader.
+        The predictions are in the original state as they come from the model, i.e. not transformed.
+        """
+
+        self.model.eval()
+
+        y_hat_list = []
+
+        with torch.no_grad():
+            for batch in dataloader:
+                
+                x_support = batch['x_support'].to(self.cfg.device)
+                y_support = batch['y_support'].to(self.cfg.device)
+                x_query = batch['x_query'].to(self.cfg.device)
+                
+                y_hat = self.model(x_support, y_support, x_query)
+
+                match self.cfg.task:
+                    case Task.REGRESSION:
+                        y_hat = y_hat[0, :, 0]
+                    case Task.CLASSIFICATION:
+                        y_hat = y_hat[0, :, :self.n_classes]
+
+                y_hat_list.append(einops.asnumpy(y_hat))
+
+        y_hat = np.concatenate(y_hat_list, axis=0)
+        return y_hat
+
+    def load_params(self, path):
+        self.model.load_state_dict(torch.load(path))
```

Comment:
We should include `# nosec B614` to exclude it from bandit check if we're confident that this path can't be hijacked.

Reply:
removed this code